pkg/reaper/nodereaper/nodereaper_test.go

```diff
@@ -420,7 +445,52 @@ func createFakeAwsAuth(a FakeASG, i []*ec2.Instance) ReaperAwsAuth {
 
 func (u *ReaperUnitTest) Run(t *testing.T, timeTest bool) {
 	awsAuth := createFakeAwsAuth(u.InstanceGroup, u.FakeInstances)
-	createFakeNodes(u.Nodes, u.FakeReaper)
+	createFakeNodes(u.Nodes, u.FakeReaper, []SkipLabel{})
+	createFakeEvents(u.Events, u.FakeReaper)
+	start := time.Now()
+	runFakeReaper(u.FakeReaper, awsAuth)
+	secondsSince := int(time.Since(start).Seconds())
+
+	if timeTest {
+		if secondsSince != u.ExpectedDurationSeconds {
+			t.Fatalf("expected Duration: %vs, got: %vs", u.ExpectedDurationSeconds, secondsSince)
+		}
+		return
+	}
+
+	if len(u.FakeReaper.UnreadyNodes) != u.ExpectedUnready {
+		t.Fatalf("expected Unready: %v, got: %v", u.ExpectedUnready, len(u.FakeReaper.UnreadyNodes))
+	}
+
+	if len(u.FakeReaper.DrainableInstances) != u.ExpectedDrainable {
+		t.Fatalf("expected Drainable: %v, got: %v", u.ExpectedDrainable, len(u.FakeReaper.DrainableInstances))
+	}
+
+	if len(u.FakeReaper.ReapableInstances) != u.ExpectedReapable {
+		t.Fatalf("expected Reapable: %v, got: %v", u.ExpectedReapable, len(u.FakeReaper.ReapableInstances))
+	}
+
+	if len(u.FakeReaper.AgeDrainReapableInstances) != u.ExpectedOldReapable {
+		t.Fatalf("expected Age Reapable: %v, got: %v", u.ExpectedOldReapable, len(u.FakeReaper.AgeDrainReapableInstances))
+	}
+
+	if u.FakeReaper.DrainedInstances != u.ExpectedDrained {
+		t.Fatalf("expected Drained: %v, got: %v", u.ExpectedDrained, u.FakeReaper.DrainedInstances)
+	}
+
+	if u.FakeReaper.TerminatedInstances != u.ExpectedTerminated {
+		t.Fatalf("expected Terminated: %v, got: %v", u.ExpectedTerminated, u.FakeReaper.TerminatedInstances)
+	}
+	if len(u.ExpectedKillOrder) != 0 {
+		if !reflect.DeepEqual(u.FakeReaper.AgeKillOrder, u.ExpectedKillOrder) {
+			t.Fatalf("expected KillOrder: %v, got: %v", u.ExpectedKillOrder, u.FakeReaper.AgeKillOrder)
+		}
+	}
+}
+
+func (u *ReaperUnitTest) RunWithSkipLabel(t *testing.T, timeTest bool, skipLabels []SkipLabel) {
```

Comment:
Mm, do we need a separate method vs. `Run()`? seems like extra duplicate code here.
Can we just use `Run()` and add `Labels` as part of type `ReaperUnitTest`? then you can create nodes with the extra argument if `len(u.Labels) != 0`

Reply:
I wanted to minimize touching the existing tests and went with this implementation as there isn't method overloading in Go. But editing the `ReaperUnitTest` does seem cleaner. I'll update that.